ViAn/GUI/projectwidget.cpp

```diff
@@ -678,26 +688,71 @@ void ProjectWidget::remove_item() {
     QString info_text = "Do you wish to continue?";
     if (message_box(text, info_text, true)) {
         for (auto item : selectedItems()) {
-            if (item->type() == FOLDER_ITEM) {
-                std::vector<VideoItem*> v_items;
-                get_video_items(item, v_items);
-                for (auto v_item : v_items) {
-                    emit item_removed(v_item->get_video_project());
-                }
-            }
+            remove_tree_item(item);
+        }
+    }
+}
 
-            else if (item->type() == VIDEO_ITEM) {
-                VideoItem* vid_item = dynamic_cast<VideoItem*>(item);
-                emit item_removed(vid_item->get_video_project());
-            }
-            // TODO Fix these cases
-//            else if (item->type() == TAG_ITEM || item->type() == ANALYSIS_ITEM) {
-//
-//            }
-            delete item;
+/**
+ * @brief ProjectWidget::remove_tree_item
+ * @param item
+ * Recursive function for removing item and all of its children
+ */
+void ProjectWidget::remove_tree_item(QTreeWidgetItem* item) {
```

Comment:
Quite a large function, should probably be refactored into function for each item type. Should also use a switch instead of if